### PR TITLE
Proxy introduce allowed ips

### PR DESCRIPTION
**Pull request by khanova**
## Problem

Proxy doesn't accept wake_compute responses with the allowed IPs.

## Summary of changes

Extend wake_compute api to be able to return allowed_ips.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: proxy/src/console/messages.rs [hunk 1]
```diff
@@ -28,6 +28,7 @@ impl fmt::Debug for GetRoleSecret {
 pub struct WakeCompute {
     pub address: Box<str>,
     pub aux: MetricsAuxInfo,
+    pub allowed_ips: Option<Vec<Box<str>>>,
```
Comment (khanova) [inline proxy/src/console/messages.rs:31]: I noticed that we store this responses to the cache.
Would it make sense to cache allowed_ips as well?

Comment (khanova) [reply proxy/src/console/messages.rs:31]: Ok, I see, that ttl is quite small, so it's fine.

Comment (conradludgate) [reply proxy/src/console/messages.rs:31]: Hmm. Do we want this on wake_compute or on get_auth_info

Comment (khanova) [reply proxy/src/console/messages.rs:31]: Hm, I think, the original suggestion was to add it to `wake_compute` (https://github.com/neondatabase/neon/issues/5628#issuecomment-1781420603).

But, probably, you are right and get_auth_info would make more sense.

**Review by conradludgate — APPROVED** [review]

**Pull request merged**